vfs/http_streambuf.cc

```diff
@@ -315,18 +329,19 @@ struct HttpStreamingDownloadSource {
     {
         impl.reset();
     }
+
 };
 
-std::pair<std::unique_ptr<std::streambuf>, FsObjectInfo>
+std::pair<std::unique_ptr<std::streambuf>, shared_ptr<FsObjectInfo>>
```

Comment:
I think I need to, otherwise it copy constructs a new one before the exception is set.